### DISCUSSION TITLE
Possible to pair without wifi credentials?

### DISCUSSION THREAD

**Discussion by Snipeye** [HomeSpan/HomeSpan#321] [Q&A] [answered]
Hi!

I only recently started getting into homekit, and realized there are some things that I would love to try building.

I was very happy to find this library, and I'm making preparations to use it.  I wanted to check my understanding of the pairing functionality before I dig too far into things.  For reference, my home hub is a HomePod Mini.

On my smart light switches I recently bought, I wired them into the wall (so they'd have power), then went to my "home" app on my phone.  When attempting to add a new device, I could see them there automatically and add them into my home - I didn't have to configure any wifi credentials.  I'm guessing this is something the switches are doing to advertise themselves to make it easy/convenient.

Does HomeSpan offer this functionality?  From what I was reading in the documentation, it seemed like I needed to either use the CLI through serial or join the ESP's AP in order to add my wifi credentials so the device could connect.

My thought is that perhaps this is a limitation of the non-MFi HomeKit specification?

Thanks!

**Comment by HomeSpan** [comment, marked as answer]
The ability for a device to autodetect WiFi Credentials via your iPhone is limited by Apple to commercial devices only.  However, there are numerous ways of entering and storing your WiFi Credentials via HomeSpan - See this [FAQ](https://github.com/HomeSpan/HomeSpan/blob/master/docs/FAQ.md#how-do-i-set-my-wifi-ssid-and-password) as well as the [API](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md) for details.

**Reply by Snipeye** [reply]
Is that what's going on under the hood?  I guess that makes sense.  It also seems that HomeSpan only supports wifi backend, not BLE?

Thank you so much!

**Reply by HomeSpan** [reply]
That's correct - HomeSpan only support WiFi, not BLE.  HomeKit BLE is completely separate protocol, so its a heavy lift to add in.

**Reply by Snipeye** [reply]
That makes a bunch of sense.  Thank you for the information - and for all the effort you've put in to making this!  I can't wait to give it a shot.
